data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/MultipartCatalogDescriptionRequestSender.java

```diff
@@ -84,44 +85,23 @@ protected Message buildMessageHeader(CatalogRequest request, DynamicAttributeTok
     }
 
     @Override
-    protected Catalog getResponseContent(IdsMultipartParts parts) throws Exception {
-        ObjectMapper objectMapper = getObjectMapper();
-
+    protected Catalog getResponseContent(IdsMultipartParts parts) {
         if (parts.getPayload() == null) {
             throw new EdcException("Payload was null but connector self-description was expected");
         }
 
-        BaseConnector baseConnector;
-        try {
-            baseConnector = objectMapper.readValue(parts.getPayload().readAllBytes(), BaseConnector.class);
-        } catch (IOException exception) {
-            throw new EdcException(String.format("Could not deserialize connector self-description: %s", exception.getMessage()));
+        BaseConnector baseConnector = getBaseConnector(getObjectMapper(), parts);
+        if (baseConnector.getResourceCatalog() == null || baseConnector.getResourceCatalog().isEmpty()) {
```

Comment:
Wouldn't it be valid to have no resources, e.g. if the client does not have access to any resources or the current connector does not have any configured?

Reply:
Yes but what do we do in this case?
If you look at the current main code, if `baseConnector.getResourceCatalog().isEmpty()`, then it would lead `resourceCatalog` to be null and the transformer would be called with a null value while this is not allowed (input argument is annoted with @NotNull).
My concern was then to cover this case which does not seem cover in the present implem and could lead to NullPointerException.

Reply:
@bscholtes1A it should be initialized to an empty list, which will be passed to the transformer

Reply:
I updated so that now `resourceCatalog` gets an empty catalog if none is present in the base connector:
`ResourceCatalog resourceCatalog = baseConnector.getResourceCatalog().stream()
                .findFirst()
                .orElse(new ResourceCatalogBuilder().build());`

@jimmarino  could you please confirm this is what you expected?
If yes I guess we can merge this one